plugins/synaptics-rmi/fu-plugin-synaptics-rmi.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-synaptics-rmi-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_SUPPORTS_PROTOCOL, "com.synaptics.rmi");
+	fu_plugin_add_udev_subsystem (plugin, "hidraw");
+}
+
+gboolean
+fu_plugin_update_detach (FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	if (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER))
+		return TRUE;
+	locker = fu_device_locker_new (device, error);
+	if (locker == NULL)
+		return FALSE;
+	return fu_device_detach (device, error);
+}
+
+gboolean
+fu_plugin_update_attach (FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new (device, error);
+	if (locker == NULL)
+		return FALSE;
+	return fu_device_attach (device, error);
+}
+
+gboolean
+fu_plugin_udev_device_added (FuPlugin *plugin, FuUdevDevice *device, GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(FuSynapticsRmiDevice) dev = NULL;
+
+	/* interesting device? */
+	if (g_strcmp0 (fu_udev_device_get_subsystem (device), "hidraw") != 0)
+		return TRUE;
+
+	/* synaptics */
+	if (fu_udev_device_get_vendor (device) != FU_SYNAPTICS_RMI_DEVICE_VID)
```

Comment:
Considering what happened with Wacom recently - can you confirm to Synaptics there is only this one VID?

Reply:
Hmm -- maybe we just remove this check completely and rely on the quirk?

Reply:
Well I think it's ideal to not have to fill this plugin's quirk file up with quirk metadata if we can avoid it.

Reply:
I've removed the check and relying on the instance ID now.